src/components/CowCard/Subheader/Subheader.js

```diff
@@ -46,28 +46,21 @@ const Subheader = ({
   handleCowAutomaticHugChange,
   handleCowBreedChange,
   huggingMachinesRemain,
-  id,
+  id, // Player ID, not cow ID
   isCowPurchased,
 }) => {
   const numberOfFullHearts = cow.happiness * 10
   const isInBreedingPen = isCowInBreedingPen(cow, cowBreedingPen)
-  const isBreedingPenFull =
-    cowBreedingPen.cowId1 !== null && cowBreedingPen.cowId2 !== null
+  const isRoomInBreedingPen =
+    cowBreedingPen.cowId1 === null || cowBreedingPen.cowId2 === null
+  const isThisCowOfferedForTrade = cowIdOfferedForTrade === cow.id
 
-  const isCowOfferedForTrade = !!cowInventory.find(
-    ({ id }) => id === cowIdOfferedForTrade
-  )
```

Comment:
This wasn't causing the issue that was reported, but I believe this logic is wrong and led to a broken and confusing UX. This deleted code is checking to see if _any_ cow is in the breeding pen, which is irrelevant. It should check if _the cow being rendered_ is in the breeding pen, which it is now with this change. I made a number of other changes in this file to simplify the logic, as it was hard to follow previously.

Reply:
`isCowOfferedForTrade` looks like it could have been causing the bug if `id` has always been "player id" in reality. cause it would potentially never find the cow in the pen based off that. right?